api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -383,6 +384,29 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
       Workspace workspace) {
     org.pmiops.workbench.db.model.Workspace dbWorkspace = workspaceService.getRequired(
         workspaceNamespace, workspaceId);
+    String userAccess;
+    try {
+       userAccess = fireCloudService.getWorkspace(
+          workspaceNamespace, workspaceId).getAccessLevel();
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      if (e.getCode() == 404) {
+        throw new NotFoundException(String.format("Workspace %s/%s not found",
+            workspaceNamespace, workspaceId));
+      } else {
+        throw new ServerErrorException(e.getResponseBody());
+      }
+    }
+    if (userAccess == null) {
+      //Actual message.
+      throw new NotFoundException(String.format(
+          "Workspace %s/%s doesn't exist",
+          workspaceNamespace, workspaceId));
+    }
+
+    if (userAccess.equals(WorkspaceAccessLevel.READER.toString())) {
```

Comment:
It might be better to whitelist OWNER and WRITER here, instead of checking for READER... that way if we add some new role we won't assume it has write permission.